### DISCUSSION TITLE
why in some example you call ts.dispose?

### DISCUSSION THREAD

**Discussion by shurshilov** [vladmandic/human#414] [Q&A] [answered]
why in some example do you use :
```
 const result = await human.detect(tensor, myConfig);
  human.tf.dispose(tensor);
```

**Comment by vladmandic** [comment, marked as answer]
if you're only dealing with images, human will create and dispose tensors automatically.
but if you create tensor manually and use that, you need to also dispose it manually or you'll have a memory leak.
tensors are not subject to javascript automatic scope monitoring and garbage collection.

**Reply by shurshilov** [reply]
We have 6 types
export type Input = Tensor | AnyCanvas | AnyImage | AnyVideo | ImageObjects | ExternalCanvas;
Thank and only tensor should be dispose? and 5 types not need?

**Reply by vladmandic** [reply]
Correct.
